plugins/experimental/ts_lua/ts_lua.c

```diff
@@ -140,13 +141,15 @@ TSRemapDoRemap(void *ih, TSHttpTxn rh, TSRemapRequestInfo *rri)
 
   http_ctx = ts_lua_create_http_ctx(main_ctx, instance_conf);
 
-  http_ctx->txnp                = rh;
-  http_ctx->client_request_bufp = rri->requestBufp;
-  http_ctx->client_request_hdrp = rri->requestHdrp;
-  http_ctx->client_request_url  = rri->requestUrl;
-  http_ctx->rri                 = rri;
-  http_ctx->remap               = 1;
-  http_ctx->has_hook            = 0;
+  http_ctx->txnp     = rh;
+  http_ctx->has_hook = 0;
+  http_ctx->remap    = remap;
```

Comment:
It looks like you don't really need the ``remap`` variable in the context; is it enough to have a non-NULL rri?